lib/rubocop/cop/rails/http_status.rb

```diff
@@ -0,0 +1,168 @@
+# frozen_string_literal: true
+
+require 'rack/utils'
+
+module RuboCop
+  module Cop
+    module Rails
+      # Enforces use of symbolic or numeric value to define HTTP status.
+      #
+      # @example `EnforcedStyle: symbolic` (default)
+      #   # bad
+      #   render :foo, status: 200
+      #   render json: { foo: 'bar' }, status: 200
+      #   render plain: 'foo/bar', status: 304
+      #   redirect_to root_url, status: 200
+      #
+      #   # good
+      #   render :foo, status: :ok
+      #   render json: { foo: 'bar' }, status: :ok
+      #   render plain: 'foo/bar', status: :not_modified
+      #   redirect_to root_url, status: :moved_permanently
+      #
+      # @example `EnforcedStyle: numeric`
+      #   # bad
+      #   render :foo, status: :ok
+      #   render json: { foo: 'bar' }, status: :not_found
+      #   render plain: 'foo/bar', status: :not_modified
+      #   redirect_to root_url, status: :moved_permanently
+      #
+      #   # good
+      #   render :foo, status: 200
+      #   render json: { foo: 'bar' }, status: 404
+      #   render plain: 'foo/bar', status: 304
+      #   redirect_to root_url, status: 301
+      #
+      class HttpStatus < Cop
+        include ConfigurableEnforcedStyle
+
+        def_node_matcher :http_status, <<-PATTERN
+          (send nil? {:render :redirect_to}
+            _
+            (hash
+              (pair
+                (sym :status)
+                ${int sym})))
+        PATTERN
+
+        def_node_matcher :http_status_hash, <<-PATTERN
```

Comment:
I haven't been able to handle all cases in one `def_node_matcher` so if anyone knows a better way to do it I would like to refactor it !

Reply:
You can use "or" syntax (`{}`). For example:

```ruby
<<-PATTERN
  {
    # http_status pattern
    # http_status_hash pattern
  }
PATTERN
```

Maybe we can write it more simply if "zero or one" operator is available, but we don't have it. 

```ruby
# If "zero or one" operator is available...
<<-PATTERN
(send nil? {:render :redirect_to}
   ZERO_OR_ONE(_)
   (hash
     (pair
       (sym :status)
       ${int sym})))
PATTERN
```

Reply:
How does work operator ? Is this something rubocop should implement or is related to ruby parser ?